server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -452,6 +467,14 @@ public void onFailure(Exception e) {
             super.onFailure(e);
         }
 
+        private boolean assertNotExecuted() {
+            if (Assertions.ENABLED) {
```

Comment:
We can't get here if assertions are disabled I think

Reply:
Right, thanks! e62d927eef846b6337eec8a7272957c75a076bb2